src/tools/compiletest/src/common.rs

```diff
@@ -645,12 +655,26 @@ pub struct TargetCfg {
     pub(crate) xray: bool,
     #[serde(default = "default_reloc_model")]
     pub(crate) relocation_model: String,
+    #[serde(default)]
+    max_atomic_width: Option<u64>,
+    #[serde(default)]
+    min_atomic_width: Option<u64>,
+    #[serde(default = "default_atomic_cas")]
+    atomic_cas: bool,
```

Comment:
Problem [DETECTION 1/2]: I think this isn't quite right, we should be looking for `target_has_atomic` for the given target from information enriched via `--print=cfg`.

```bash
$ rustc --print=cfg --target=x86_64-unknown-linux-gnu
[.. irrelevant entries trimmed ..]
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
```

e.g. if you compare that to

```bash
$ rustc --print=cfg --target=aarch64-unknown-linux-gnu
[.. irrelevant entries trimmed ..]
target_has_atomic="128"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
```

which AFAIK is what #84039 was originally looking for, as the std atomic types will be available or not based on `#[cfg(target_has_atomic=...)]`.

Reply:
I would prefer that we match what `--print=cfg` produces here, because that's what this directive corresponds to, in the event that there's any kind of inconsistency.

Note that `TargetCfg` AFAIK is also enriched by `--print=cfg`, e.g. https://github.com/rust-lang/rust/blob/c49a687d636e3464cdc6599fb562c29470526b01/src/tools/compiletest/src/common.rs#L576